### ISSUE TITLE
Request for C4MIP-LPJ-GUESS output which is not part of the CMIP6  request

### ISSUE THREAD

**Issue by treerink** [EC-Earth/ece2cmor3#546]
Currently there are two tier 1 C4MIP experiments in the control output files:
``` 
cmip6-experiment-C4MIP-1pctCO2-bgc
cmip6-experiment-C4MIP-esm-ssp585
```
A request from the C4MIP - LPJ-GUESS folks is whether it is possible to output at any time two small lists of variables (one of tier 1 priority variables and one of tier 2 priority variables), so even when the CMIP6 data request does not ask for this variables in these experiments.

C4MIP - LPJ-GUESS folks 

The list of priority tier 1 variables to be included in any case:
```
cVeg           (cVeg=cLeaf+cStem+cRoot+cOther)
cLitter        (cLitter=cLitterCwd+cLitterSurf+clitterSubSurf)
cSoil
cProduct
cLand          (cLand=cVeg+cLitter+cSoil+cProduct)
```
The list of priority tier 2 variables to be included in any case:
```
cLeaf
cStem
cRoot
cOther
cLitterCwd
cLitterSurf
cLitterSubSurf
```
This concerns LPJ-GIUESS output, and this output should be included in the cmorisation as well.

**Comment by etiennesky** [comment]
Hi- the same applies to all CMIP experiments with the EC-Earth-CC model.

Also the same variables with a n prefix e.g. nLeaf, nLitterCwd need to be added.

**Comment by treerink** [comment]
Hi Etienne, 

You are making it even more complicated. CMIP for EC-Earth-CC requires a joint data request from C4MIP,CDRMIP,CMIP,LUMIP,OMIP. Now you want to request, in that joint request for each of the CMIP experiments for the EC-Earth-CC configuration, at any time the two lists in https://github.com/EC-Earth/ece2cmor3/issues/546#issue-524521207as well, right?

**Comment by treerink** [comment]
Etienne, can you please list this n prefix variables. So I am sure what we are talking about (often lists are _almost_ the same).

**Comment by etiennesky** [comment]
> Hi Etienne,
> 
> You are making it even more complicated. CMIP for EC-Earth-CC requires a joint data request from C4MIP,CDRMIP,CMIP,LUMIP,OMIP. Now you want to request, in that joint request for each of the CMIP experiments for the EC-Earth-CC configuration, at any time the two lists in [#546 (comment)](https://github.com/EC-Earth/ece2cmor3/issues/546#issue-524521207as) well, right?

there is no point in having these variables only for the C4MIP specific experiments (cmip6-experiment-C4MIP-1pctCO2-bgc, cmip6-experiment-C4MIP-esm-ssp585), we need them for all experiment e.g. piControl, historical.

**Comment by etiennesky** [comment]
> Etienne, can you please list this n prefix variables. So I am sure what we are talking about (often lists are _almost_ the same).

here you go (they are exactly the same)

nVeg, nLitter, nSoil, nProduc, nLand, nLeaf, nStem, nRoot, nOther, nLitterCwd, nLitterSurf, nLitterSubSurf

@nierad can you please add David to this issue?

There are also a number of fluxes which are missing, but I think we can use another issue for that.

**Comment by nierad** [comment]
@etiennesky I sent him an email.

**Comment by warlind** [comment]
As Etienne mentions this is the case for all EC-Earth-CC simulations. And to be totally honest I would say that we (LPJ-GUESS) group would like all of these variables to always be outputted if any versions of EC-Earth is used with LPJ-GUESS active. 

So the full list of variables that we would like to ALWAYS have is the following:

CARBON POOLS
cVeg           (cVeg=cLeaf+cStem+cRoot+cOther)
cLitter        (cLitter=cLitterCwd+cLitterSurf+clitterSubSurf)
cSoil
cProduct
cLand          (cLand=cVeg+cLitter+cSoil+cProduct)

cLeaf
cStem
cRoot
cOther
cLitterCwd
cLitterSurf
cLitterSubSurf

CARBON FLUXES
npp
gpp
ra
rh
fFireNat
fCLandToOcean
fVegOther
fProductDecomp
nep
fLuc
nbp


NITROGEN POOLS
nVeg           (nVeg=nLeaf+nStem+nRoot+nOther)
nLitter        (nLitter=nLitterCwd+nLitterSurf+nlitterSubSurf)
nSoil
nMineral
nProduct
nLand          (nLand=nVeg+nLitter+nSoil+nMineral+nProduct)

nLeaf
nStem
nRoot
nOther
nLitterCwd
nLitterSurf
nLitterSubSurf

NITROGEN FLUXES
fBNF
fNdep
fNfert
fNnetmin
fNup
fNloss		(fnloss=fNLandToOcean+different gases)
fNLandToOcean

**Comment by treerink** [comment]
You list a long list of variables, but we need to know table - variable combinations. Is this only concerning the `Lmon` table?

You want this for all EC-Earth3 configurations, thus for:
```
EC-EARTH3-CC
EC-EARTH3-Veg
EC-EARTH3-Veg-LR
```
I hope you realise this is rather complicated. Ece2cmor3 is totally based on following the data request, deviating from that for specific cases is error prone and makes the end result intransparent.

Anyway, to make progress now, we need to know which tables are involved!

**Comment by treerink** [comment]
Probably also Emon?

**Comment by treerink** [comment]
Hi David, Etienne,

For part one we seem to have a solution. I upload here a json file (the extension is changed to txt because github does not allow the json extension to upload here). Can you modify this file such that all table - variable combinations which you want to output permanently with LPJ-GUESS remain in this file.  If you can do this now, we might be able to get this issue in v1.3.0.

[permanent-tasks.txt](https://github.com/EC-Earth/ece2cmor3/files/3879324/permanent-tasks.txt)

**Comment by etiennesky** [comment]
Hi I will look into this.

I forgot also to request to include burntAreaAl in Amon table, recent changes to lpjg (in a MR from today) allow to cmorize it.

This one is in the data request but is missing from the lpjg output files and json files (maybe it was an ignored variable)?

**Comment by treerink** [comment]
Ok we have technically a solution, also for part two now: the json data request file for cmorising. So as soon you have the permanent-task list available we can go ahead. It then needs a rerun of `genecec` and updating the `r7144-control-output-files` branch.

**Comment by etiennesky** [comment]
awesome! I will get back to you. also , can you please make the changes for pices mentionned in #521 before running genecec ?

**Comment by etiennesky** [comment]
Looks good - I would add burntFractionAll to Lmon - unless it is now produced by genecec

The following remarks:

- not sure these are necessary : cLitterAbove cLitterBelow 
- many *Lut files  are not in the c4mip request (e.g. gppLut in Emon) - maybe these are from lumip? 
- files in Eyr are not in the c4mip data request
- these are missing from data request and clip service is down so I cannot search for them :  fVegOther nep fLuc nbp

**Comment by warlind** [comment]
cLitterSurf and clitterSubSurf only necessary if we are going to split cLitter

cLitter=cLitterCwd+cLitterSurf+clitterSubSurf

nbp - lmon
nep - emon
fLuc - emon
fVegOther - doesn't exist in the request. Needed to close LPJ-GUESS carbon balance.

**Comment by etiennesky** [comment]
Hi @warlind about fVegOther, is this value large, and could we instead add it to an existing f* variable? I am not sure it would be suitable to output a variable which doesn't exist.

**Comment by treerink** [comment]
Hi Etienne & David,

I uploaded as a starting point this file:
`ece2cmor3/ece2cmor3/resources/permanent-tasks.json`
I took just something which includes a lot of LPJ-GUESS variables, so you have a clear format and just can delete many of them and add a few ones which might be missing here. 

What I expect to get back from you now, to be able to finish this issue and to be able to rerun `genecec,` is an update of this file such that it includes the variables mentioned in https://github.com/EC-Earth/ece2cmor3/issues/546#issuecomment-556988391 with the correct tables. So I expect that you delete about 80% of the variables from the current `ece2cmor3/ece2cmor3/resources/permanent-tasks.json` and keep just those mentioned in https://github.com/EC-Earth/ece2cmor3/issues/546#issuecomment-556988391 for the tables you are interested in.

If you need further help, let me know.

**Comment by warlind** [comment]
fVegOther will be all leaf and root C mass times (1-phen), so during winter for summergreens (phen=0) it will be a of leaf and root C mass. Then during the growing season, it will be the sum of the yearly NPP. On top of this a couple of other things as well. So It will be big.

**Comment by treerink** [comment]
With a lack of progress here I took post https://github.com/EC-Earth/ece2cmor3/issues/546#issuecomment-556988391 and marked those with an x in the `permanent-tasks.json` file ((making it invalid json of course). Only the variable `fVegOther` did not have a hit. Following this list those marked with an x at the start of the line will be always saved in the output, see the diff below:

```diff
diff --git a/ece2cmor3/resources/permanent-tasks.json b/ece2cmor3/resources/permanent-tasks.json
index f377484..c5cc132 100644
--- a/ece2cmor3/resources/permanent-tasks.json
+++ b/ece2cmor3/resources/permanent-tasks.json
@@ -14,22 +14,22 @@
     "Emon": [
         "gppLut",
         "fLulccAtmLut",
-        "cOther",
-        "fBNF",
+x        "cOther",
+x        "fBNF",
         "fDeforestToAtmos",
         "fNgas",
-        "nStem",
-        "nLitterCwd",
-        "fNfert",
-        "nLitterSurf",
+x        "nStem",
+x        "nLitterCwd",
+x        "fNfert",
+x        "nLitterSurf",
         "fLitterFire",
         "treeFracBdlDcd",
         "treeFracNdlEvg",
         "rhLitter",
-        "nRoot",
-        "fLuc",
+x        "nRoot",
+x        "fLuc",
         "gppTree",
-        "nLitterSubSurf",
+x        "nLitterSubSurf",
         "fHarvestToAtmos",
         "nMineralNH4",
         "raStem",
@@ -38,34 +38,34 @@
         "raLut",
         "nppTree",
         "fracLut",
-        "nLeaf",
+x        "nLeaf",
         "netAtmosLandCO2Flux",
         "raTree",
         "grassFracC4",
-        "cLitterSubSurf",
-        "nep",
+x        "cLitterSubSurf",
+x        "nep",
         "grassFracC3",
         "cropFracC3",
         "fNVegLitter",
         "mrroLut",
-        "fNdep",
+x        "fNdep",
         "raLeaf",
         "vegFrac",
         "rhSoil",
-        "cSoil",
-        "fNnetmin",
-        "fCLandToOcean",
-        "cLitterSurf",
-        "cStem",
+x        "cSoil",
+x        "fNnetmin",
+x        "fCLandToOcean",
+x        "cLitterSurf",
+x        "cStem",
         "fDeforestToProduct",
         "fNgasFire",
         "fNProduct",
         "fNLitterSoil",
         "fNAnthDisturb",
-        "fFireNat",
+x        "fFireNat",
         "treeFracNdlDcd",
         "fProductDecompLut",
-        "nMineral",
+x        "nMineral",
         "fNgasNonFire",
         "raGrass",
         "fFireAll",
@@ -73,56 +73,56 @@
         "prCrop",
         "laiLut",
         "fVegFire",
-        "nLand",
+x        "nLand",
         "fLulccResidueLut",
-        "nVeg",
+x        "nVeg",
         "mrsosLut",
         "raRoot",
         "mrsoLut",
         "nMineralNO3",
         "treeFracBdlEvg",
-        "nSoil",
+x        "nSoil",
         "fLulccProductLut",
         "nppGrass",
         "rhLut",
-        "cLitterCwd",
+x        "cLitterCwd",
         "irrLut",
-        "nProduct",
+x        "nProduct",
         "vegHeightTree",
-        "nLitter",
+x        "nLitter",
         "cWood",
         "cTotFireLut",
-        "fNloss",
+x        "fNloss",
         "fNleach",
         "necbLut",
-        "fNup",
-        "nOther",
+x        "fNup",
+x        "nOther",
         "gppGrass",
-        "fNLandToOcean",
+x        "fNLandToOcean",
         "evspsblpot",
         "nppLut",
         "pastureFracC3",
         "pastureFracC4",
-        "fProductDecomp",
-        "cLand",
+x        "fProductDecomp",
+x        "cLand",
         "fAnthDisturb"
     ],
     "Eyr": [
         "fracLut",
-        "cProduct",
+x        "cProduct",
         "cropFrac",
         "residualFrac",
         "treeFrac",
         "baresoilFrac",
-        "cVeg",
+x        "cVeg",
         "shrubFrac",
         "cSoilLut",
         "fracInLut",
-        "cLitter",
+x        "cLitter",
         "cVegLut",
         "grassFrac",
         "cLitterLut",
-        "cSoil",
+x        "cSoil",
         "cProductLut",
         "fracOutLut",
         "vegFrac"
@@ -132,30 +132,30 @@
         "cropFrac",
         "shrubFrac",
         "prveg",
-        "nbp",
-        "cProduct",
+x        "nbp",
+x        "cProduct",
         "fGrazing",
-        "cLeaf",
+x        "cLeaf",
         "fHarvest",
         "cLitterAbove",
-        "rh",
+x        "rh",
         "cSoilMedium",
         "fFire",
         "baresoilFrac",
-        "cVeg",
-        "cRoot",
+x        "cVeg",
+x        "cRoot",
         "grassFrac",
-        "ra",
-        "npp",
+x        "ra",
+x        "npp",
         "cCwd",
-        "gpp",
+x        "gpp",
         "rMaint",
         "treeFrac",
         "evspsblsoi",
         "lai",
         "residualFrac",
         "cSoilSlow",
-        "cLitter",
+x        "cLitter",
         "evspsblveg",
         "cSoilFast",
         "pastureFrac",
```

I must say that in the end it looks like we are first configuring each instruction file for each MIP experiment in order to save output (this has been a lot of work earlier), and now we are overruling that by forcing to output a lot permanently. Is there a broad agreement about this, does this need a further discussion in the CMIP6 WG?

`Lmon burntFractionAll` is part of the ignored file, marked with Davids comment: `Can not be produced by LPJ-GUESS: Not calculated in LPJ-GUESS`

What is your conclusion about `cLitterSurf and clitterSubSurf`?

**Comment by warlind** [comment]
I don't remember why I made that comment about burntFractionAll, but at the moment we do output it so it could also be on the ALWAYS list. cLitterSurf and clitterSubSurf we don't need I would say, as we don't really distinguish between surface and soil in this version of LIP-GUESS.

**Comment by treerink** [comment]
Ok, see your comment just after I finished all. The identification status of `burntFractionAll` has then to be changed, and all related stuff. It is only about `Lmon burntFractionAll`, right? (Makes it easier to check all is fine).

Ok I can take off `cLitterSurf` and `clitterSubSurf` from the permanent list again, that's easy.

**Comment by treerink** [comment]
I have run `genecec` for the changes of this issue, the produced control output files are uploaded in the  `r7144-control-output-files` branch (see also [ece -earthortal issue 694](https://dev.ec-earth.org/issues/694)). Can you test a 1 year run with EC-Earth3-Veg of CC and cmorise those results?

For here I now close the issue.

**Comment by etiennesky** [comment]
> I must say that in the end it looks like we are first configuring each instruction file for each MIP experiment in order to save output (this has been a lot of work earlier), and now we are overruling that by forcing to output a lot permanently. Is there a broad agreement about this, does this need a further discussion in the CMIP6 WG?

Hi @treerink about outputting extra fields (and not saving space), I would argue that without these changes the output is not complete, and the additional space used will be small, since these are 2d vars.
> 
> `Lmon burntFractionAll` is part of the ignored file, marked with Davids comment: `Can not be produced by LPJ-GUESS: Not calculated in LPJ-GUESS`


burnFractionAll is now output at monthly frequency since yesterday's merge into trunk, it did not work before (it was in yearly, not monthly files).

> 
> What is your conclusion about `cLitterSurf and clitterSubSurf`?

**Comment by etiennesky** [comment]
> I don't remember why I made that comment about burntFractionAll, but at the moment we do output it so it could also be on the ALWAYS list. cLitterSurf and clitterSubSurf we don't need I would say, as we don't really distinguish between surface and soil in this version of LIP-GUESS.

are you sure about that? There is clearly code for outputting them via mcLitterSurf mcLitterSubSurf mcLitterCwd in miscoutput.cpp

Also - the corresponding variable for nitrogen (nLitterSurf, nLitterSubSurf) are still there.

**Comment by warlind** [comment]
Of course, all of them are there. It is just if they have to be in the list of permanent variables. If you feel that they are important then they could stay in the permanent list, but with litterCwd and litter then we know the sum of them. but it might be that you need the surf fraction for your fire study.

**Comment by etiennesky** [comment]
> Of course, all of them are there. It is just if they have to be in the list of permanent variables. If you feel that they are important then they could stay in the permanent list, but with litterCwd and litter then we know the sum of them. but it might be that you need the surf fraction for your fire study.

I understood from "we don't really distinguish between surface and soil in this version" that they did not exist. Only they are not considered afterwards internally.

I would prefer to output them so as to make it clear that cLitter=cLitterCwd+cLitterSurface+cLitterSubsurface, due to the confusion in the c4mip paper definition vs. the data request definition.

**Comment by treerink** [comment]
Please be hesitant with the permanent list. If you personally need a variable for a specific variable you always manually can add it to the instruction list and the json data request file. Having said that, if everybody wants the same - including a set of variables which is (often) outside the data request, then it should be in this permanent list.

I conclude that the only remaining point of discussion here is including or excluding `cLitterSurf` & `clitterSubSurf` in the permanent list. No other issues, right?

**Comment by etiennesky** [comment]
cLitterSurf / clitterSubSurf is a point to be clarified, as well as their nitrogen counterparts nLitterSurf / nlitterSubSurf which are there right now.

I understand that this can be changed manually, but because of the potential confusion in this variable, I would argue to always output it.

**Comment by treerink** [comment]
Okay locally I added `cLitterSurf` & `clitterSubSurf` again to the `permanent-tasks.json` file, and I kept `nLitterSurf` & `nlitterSubSurf` also in. Etienne happy? David ok?

I want to get this issue ready.

**Comment by warlind** [comment]
Super happy!

**Comment by treerink** [comment]
Great :)

I have run `genecec`, the changes in the produced control output files are uploaded in the  `r7144-control-output-files` branch (see also [ece -earthortal issue 694-25](https://dev.ec-earth.org/issues/694-25#note-25)). Can you test a 1 year run with EC-Earth3-Veg of CC and cmorise those results?

Close the issue here.

**Comment by etiennesky** [comment]
Thanks for the great work @treerink ! I ran 1 year with CMIP/EC-EARTH-Veg/cmip6-experiment-CMIP-piControl and the new variables are output and cmorized correctly (except for fOther)

However, fOther is not cmorized, not produced. Can we do something about this?

There is another issue I found, is that burnFractionAll is multiplied by 100 (by fract2percent conversion), but LPJG already outputs this variable in percentage, unlike the other fraction vars.

Could you make an exception for this variable in scripts/generate-lpjguess.json.sh and update resources/lpjgpar.json, removing the conversion string? Hopefully this is easier than fixing lpjg output and doing another vendor drop & merge request.

**Comment by etiennesky** [comment]
Also, the following files still have clitterAbove/Below - perhaps they have not been updated ?

```
cns61181@login2:/gpfs/scratch/cns61/cns61181/a2dr/ctrl/cmip6-output-control-files> grep -r cLitterAbove 
test-all-ece-mip-variables/ece-cmip6-data-request-varlist-all-EC-EARTH-CC.json:            "cLitterAbove",
ScenarioMIP/EC-EARTH-Veg-LR/cmip6-experiment-ScenarioMIP-ssp585/cmip6-data-request-varlist-ScenarioMIP-ssp585-EC-EARTH-Veg-LR.json:            "cLitterAbove",
ScenarioMIP/EC-EARTH-Veg-LR/cmip6-experiment-ScenarioMIP-ssp585/lpjg_cmip6_output.ins:file_cLitterAbove_monthly "cLitterAbove_monthly.out"
ScenarioMIP/EC-EARTH-Veg-LR/cmip6-experiment-ScenarioMIP-ssp245/cmip6-data-request-varlist-ScenarioMIP-ssp245-EC-EARTH-Veg-LR.json:            "cLitterAbove",
ScenarioMIP/EC-EARTH-Veg-LR/cmip6-experiment-ScenarioMIP-ssp245/lpjg_cmip6_output.ins:file_cLitterAbove_monthly "cLitterAbove_monthly.out"
ScenarioMIP/EC-EARTH-Veg-LR/cmip6-experiment-ScenarioMIP-ssp126/cmip6-data-request-varlist-ScenarioMIP-ssp126-EC-EARTH-Veg-LR.json:            "cLitterAbove",
ScenarioMIP/EC-EARTH-Veg-LR/cmip6-experiment-ScenarioMIP-ssp126/lpjg_cmip6_output.ins:file_cLitterAbove_monthly "cLitterAbove_monthly.out"
ScenarioMIP/EC-EARTH-Veg-LR/cmip6-experiment-ScenarioMIP-ssp370/cmip6-data-request-varlist-ScenarioMIP-ssp370-EC-EARTH-Veg-LR.json:            "cLitterAbove",
ScenarioMIP/EC-EARTH-Veg-LR/cmip6-experiment-ScenarioMIP-ssp370/lpjg_cmip6_output.ins:file_cLitterAbove_monthly "cLitterAbove_monthly.out"
ScenarioMIP/EC-EARTH-AerChem/cmip6-experiment-ScenarioMIP-ssp585/lpjg_cmip6_output.ins:file_cLitterAbove_monthly "cLitterAbove_monthly.out"
ScenarioMIP/EC-EARTH-AerChem/cmip6-experiment-ScenarioMIP-ssp245/lpjg_cmip6_output.ins:file_cLitterAbove_monthly "cLitterAbove_monthly.out"
ScenarioMIP/EC-EARTH-AerChem/cmip6-experiment-ScenarioMIP-ssp126/lpjg_cmip6_output.ins:file_cLitterAbove_monthly "cLitterAbove_monthly.out"
ScenarioMIP/EC-EARTH-AerChem/cmip6-experiment-ScenarioMIP-ssp370/lpjg_cmip6_output.ins:file_cLitterAbove_monthly "cLitterAbove_monthly.out"
ScenarioMIP/EC-EARTH-Veg/cmip6-experiment-ScenarioMIP-ssp585/cmip6-data-request-varlist-ScenarioMIP-ssp585-EC-EARTH-Veg.json:            "cLitterAbove",
ScenarioMIP/EC-EARTH-Veg/cmip6-experiment-ScenarioMIP-ssp585/lpjg_cmip6_output.ins:file_cLitterAbove_monthly "cLitterAbove_monthly.out"
ScenarioMIP/EC-EARTH-Veg/cmip6-experiment-ScenarioMIP-ssp245/cmip6-data-request-varlist-ScenarioMIP-ssp245-EC-EARTH-Veg.json:            "cLitterAbove",
ScenarioMIP/EC-EARTH-Veg/cmip6-experiment-ScenarioMIP-ssp245/lpjg_cmip6_output.ins:file_cLitterAbove_monthly "cLitterAbove_monthly.out"
ScenarioMIP/EC-EARTH-Veg/cmip6-experiment-ScenarioMIP-ssp126/cmip6-data-request-varlist-ScenarioMIP-ssp126-EC-EARTH-Veg.json:            "cLitterAbove",
ScenarioMIP/EC-EARTH-Veg/cmip6-experiment-ScenarioMIP-ssp126/lpjg_cmip6_output.ins:file_cLitterAbove_monthly "cLitterAbove_monthly.out"
ScenarioMIP/EC-EARTH-Veg/cmip6-experiment-ScenarioMIP-ssp370/cmip6-data-request-varlist-ScenarioMIP-ssp370-EC-EARTH-Veg.json:            "cLitterAbove",
ScenarioMIP/EC-EARTH-Veg/cmip6-experiment-ScenarioMIP-ssp370/lpjg_cmip6_output.ins:file_cLitterAbove_monthly "cLitterAbove_monthly.out"
ScenarioMIP/EC-EARTH-AOGCM/cmip6-experiment-ScenarioMIP-ssp585/lpjg_cmip6_output.ins:file_cLitterAbove_monthly "cLitterAbove_monthly.out"
ScenarioMIP/EC-EARTH-AOGCM/cmip6-experiment-ScenarioMIP-ssp245/lpjg_cmip6_output.ins:file_cLitterAbove_monthly "cLitterAbove_monthly.out"
ScenarioMIP/EC-EARTH-AOGCM/cmip6-experiment-ScenarioMIP-ssp126/lpjg_cmip6_output.ins:file_cLitterAbove_monthly "cLitterAbove_monthly.out"
ScenarioMIP/EC-EARTH-AOGCM/cmip6-experiment-ScenarioMIP-ssp370/lpjg_cmip6_output.ins:file_cLitterAbove_monthly "cLitterAbove_monthly.out"
CMIP/EC-EARTH-AOGCM/cmip6-experiment-CMIP-historical/lpjg_cmip6_output.ins:file_cLitterAbove_monthly "cLitterAbove_monthly.out"
```

**Comment by treerink** [comment]
Adhoc fix for `burnFractionAll` added.

I have no idea about `fOther`, variable seems to be not part of anything? Please describe problem.

What is the problem with `clitterAbove&`  `clitterBelow`, I see they are in `lpjg_cmip6_output.ins`, looks like correct. They are not part of the `permanent-tasks.json` file. Or do you want them to remove everywhere, so changing them to the ignored list??

**Comment by etiennesky** [comment]
> Adhoc fix for `burnFractionAll` added.
thanks!
> 
> I have no idea about `fOther`, variable seems to be not part of anything? Please describe problem.

I am sorry I meant fVegOther. It is not in the data request but needed to close the carbon budget. 

See comments https://github.com/EC-Earth/ece2cmor3/issues/546#issuecomment-557593008 and https://github.com/EC-Earth/ece2cmor3/issues/546#issuecomment-558146908

> 
> What is the problem with `clitterAbove&` `clitterBelow`, I see they are in `lpjg_cmip6_output.ins`, looks like correct. They are not part of the `permanent-tasks.json` file. Or do you want them to remove everywhere, so changing them to the ignored list??

The problem is  they appear in seemingly random lpjg_cmip6_output.ins files (just a few ScnenarioMIP and one CMIP) .  I am a bit at loss here, it seems like they are in the CMIP data request but not in the C4MIP one, where cLitterSubSurf and cLitterAbove are instead . Quite confusing.

It doesn't help that the clipc-services dreq site is down (http://clipc-services.ceda.ac.uk/dreq/mipVars.html and http://clipc-services.ceda.ac.uk/dreq/mipVars.html)  and has been so for quite a long time, do you know an alternative?

**Comment by treerink** [comment]
Also `fVegOther` is not present in any of the data requests of any MIP experiment in which EC-Earth* is participating. So I have no idea about this variable, are you sure its cmor name is `fVegOther`? Also here it doesn't help that the [clip site ](http://clipc-services.ceda.ac.uk/dreq/index.html) is down for quite a time, so I can't check. 

What is also unclear to me if you have this variable already implemented in LPJ-GUESS, otherwise if not it will cause trouble as soon it pops up in the instruction files.

**Comment by treerink** [comment]
About `clitterAbove` & `clitterBelow`: Very likely what you see reflects the data request. In all situations until now where we were confused it appears that `genecec` was always correct in following the data request. This of course doesn't say anything whether you like or agree the data request choice itself.

Because the [clip site ](http://clipc-services.ceda.ac.uk/dreq/index.html) is down, you can run the following when your `ece2cmor3` environment is active from the `ece2cmor/scripts` directory:
```
./determine-missing-variables.sh CMIP,AerChemMIP,CDRMIP,C4MIP,DCPP,HighResMIP,ISMIP6,LS3MIP,LUMIP,OMIP,PAMIP,PMIP,RFMIP,ScenarioMIP,VolMIP,CORDEX,DynVarMIP,SIMIP,VIACSAB CMIP 1 1

./determine-missing-variables.sh C4MIP       C4MIP        1 1
```
This will generate many of the original data request .xlsx files. And then you can start digging, which is quite a job. I already checked the monthly tables of the C4MIP historical request, it does not contain these two variables.

As it seems this issue is so unclear yet, I propose we leave this as it is because probably in terms of data request it is correct, in order to ship the rest of this issue in the upcoming release. Okay?

**Comment by etiennesky** [comment]
> Also `fVegOther` is not present in any of the data requests of any MIP experiment in which EC-Earth* is participating. So I have no idea about this variable, are you sure its cmor name is `fVegOther`? Also here it doesn't help that the [clip site ](http://clipc-services.ceda.ac.uk/dreq/index.html) is down for quite a time, so I can't check.
> 
> What is also unclear to me if you have this variable already implemented in LPJ-GUESS, otherwise if not it will cause trouble as soon it pops up in the instruction files.

This variable is implemented in LPJ but it is not in the data requests. It was implemented by @warlind to close the carbon budget. Can we have it added to the .ins files via this mechanism, and also in the .json files (in say the Emon table) ? We can later on decide how to cmorize it.

**Comment by treerink** [comment]
Ok that makes the `fVegOther` case clear to me. The problem is indeed that if it's not part of the data request `ece2cmor` can't cmorise it. If I for instance add it to the permanent-task list, it will be also in the json data request file, leading to an error.

At which frequency/frequencies do you have / do you want to output `fVegOther` in LPJ-GUESS?

**Comment by warlind** [comment]
LPJ-GUESS only generates fVegOther on a monthly timestep. 

There are many duplicates in the CMIP6 data request. So clitterAbove & clitterBelow are the same as cLitterSurf & clitterSubSurf just with different names. I think for the CMIP7 data request they have to do a better job with not doing things like this and making sure that the portal is in agreement with the protocol papers!

**Comment by treerink** [comment]
The order of items in the .ins files does not matter, right? So I could just add the line for `fVegOther` at the bottom?

**Comment by warlind** [comment]
Exactly, it doesn't matter where you put it.

**Comment by treerink** [comment]
Ok with an additional adhoc hack, I added `fVegOther` to **all** instruction files. Last check, this is ok, it is always added?

I run already `genecec` with this changes will upload it in the branch.  So `fVegOther` will be not part of the json data request in this way, but that is I think the best for now, because `ece2cmor3` can not handle it.

**Comment by etiennesky** [comment]
> Ok with an additional adhoc hack, I added `fVegOther` to **all** instruction files. Last check, this is ok, it is always added?
> 
> I run already `genecec` with this changes will upload it in the branch. So `fVegOther` will be not part of the json data request in this way, but that is I think the best for now, because `ece2cmor3` can not handle it.

What if it was in the data request but ece2cmor3 cannot handle it? I seem to recall that it will just give a warning message and continue, is that not still the case?

**Comment by etiennesky** [comment]
> About `clitterAbove` & `clitterBelow`: Very likely what you see reflects the data request. In all situations until now where we were confused it appears that `genecec` was always correct in following the data request. This of course doesn't say anything whether you like or agree the data request choice itself.
> 
> Because the [clip site ](http://clipc-services.ceda.ac.uk/dreq/index.html) is down, you can run the following when your `ece2cmor3` environment is active from the `ece2cmor/scripts` directory:
> 
> ```
> ./determine-missing-variables.sh CMIP,AerChemMIP,CDRMIP,C4MIP,DCPP,HighResMIP,ISMIP6,LS3MIP,LUMIP,OMIP,PAMIP,PMIP,RFMIP,ScenarioMIP,VolMIP,CORDEX,DynVarMIP,SIMIP,VIACSAB CMIP 1 1
> 
> ./determine-missing-variables.sh C4MIP       C4MIP        1 1
> ```
> 
> This will generate many of the original data request .xlsx files. And then you can start digging, which is quite a job. I already checked the monthly tables of the C4MIP historical request, it does not contain these two variables.
> 
> As it seems this issue is so unclear yet, I propose we leave this as it is because probably in terms of data request it is correct, in order to ship the rest of this issue in the upcoming release. Okay?

What is unexpected it that `clitterAbove` & `clitterBelow` do not show up in most of the .ins files, whereas if it were honoring the data request all CMIP experiments should have it.

In any case, this is probably because it is a priority 2 variable.  But why do they show up in some of the .ins files?

We could just add them to the ignore list so we don't have them anywhere.

**Comment by treerink** [comment]
Yes, obviously it is sometimes priority 1 and sometimes priority 2, that would explain it.

You really want to put them to the ignore? I would say if it is sometimes priority 1, then just deliver it?

**Comment by etiennesky** [comment]
> Yes, obviously it is sometimes priority 1 and sometimes priority 2, that would explain it.
> 
> You really want to put them to the ignore? I would say if it is sometimes priority 1, then just deliver it?

ok that is fine then, if is priority 1 then we have it, if it is 2 then not. I just didn't understand why it shows up in some experiments. thanks!

**Comment by treerink** [comment]
Great, then I believe we are there.

Closing again this issue.

**Comment by etiennesky** [comment]
thanks, once this release is over let's see how we can cmorize fVegOther

**Comment by etiennesky** [comment]
changes to lpjg .ins file crashes LPJG:

```
In /gpfs/scratch/cns61/cns61181/a2dr/19800101/fc0/runtime/run70/lpjg_cmip6_output.ins, line 91, "fvegother_monthly":
  Error: Cannot be interpreted as a number
Bad instruction file!
```

**Comment by etiennesky** [comment]
changing this line
```
file_fVegOther_monthly fVegOther_monthly.out
```
to this
```
file_fVegOther_monthly "fVegOther_monthly.out"
```
resolves the issue.

LPJG at ltast starts, still running a test to see if the output is fine

**Comment by etiennesky** [comment]
output of fVegOther looks fine, after changing the line mentioned above.

**Comment by treerink** [comment]
Ok this is solved in e5a6bfd5f72058b4f2d6fb20832170f80360f3f0. Also this needs running `genecec` and uploading the corrected results in a same new branch for EC-Earth.

**Comment by treerink** [comment]
The fixed files are uploaded in the ec-earth `r7438-control-output-files` branch.
